addons/test_mail/tests/test_mail_gateway.py

```diff
@@ -8,7 +8,9 @@
 from unittest.mock import patch
 
 from odoo import exceptions
-from odoo.addons.mail.tests.common import mail_new_test_user
+from email.message import EmailMessage
+import time
```

Comment:
We separate local imports from external/standard python libs imports. This should be under import `socket`. Imports should also be in alphabetical order as much as is reasonable.